server/src/main/java/io/crate/execution/engine/aggregation/impl/NumericSumAggregation.java

```diff
@@ -0,0 +1,330 @@
+        }
+        return state;
+    }
+
+    @Override
+    public DataType<?> partialType() {
+        return returnType;
+    }
+
+    @Override
+    public Signature signature() {
+        return signature;
+    }
+
+    @Override
+    public Signature boundSignature() {
+        return boundSignature;
+    }
+
+    @Override
+    public boolean isRemovableCumulative() {
+        return true;
+    }
+
+    @Override
+    public BigDecimal removeFromAggregatedState(RamAccounting ramAccounting,
+                                                BigDecimal previousAggState,
+                                                Input[] stateToRemove) {
+        BigDecimal value = returnType.implicitCast(stateToRemove[0].value());
+        if (value != null) {
+            if (previousAggState != null) {
+                return previousAggState.subtract(value);
+            }
+        }
+        return previousAggState;
+    }
+
+    @Override
+    public DocValueAggregator<?> getDocValueAggregator(List<DataType<?>> argumentTypes,
+                                                       List<MappedFieldType> fieldTypes) {
+        return switch (argumentTypes.get(0).id()) {
+            case ByteType.ID, ShortType.ID, IntegerType.ID, LongType.ID ->
+                new SumLong(returnType, fieldTypes.get(0).name());
+            case FloatType.ID -> new SumFloat(returnType, fieldTypes.get(0).name());
+            case DoubleType.ID -> new SumDouble(returnType, fieldTypes.get(0).name());
+            default -> null;
+        };
+    }
+
+    static class SumLong implements DocValueAggregator<OverflowAwareMutableLong> {
+
+        private final DataType<BigDecimal> returnType;
+        private final String columnName;
+        private SortedNumericDocValues values;
+
+        SumLong(DataType<BigDecimal> returnType, String columnName) {
+            this.returnType = returnType;
+            this.columnName = columnName;
+        }
+
+        @Override
+        public OverflowAwareMutableLong initialState(RamAccounting ramAccounting) {
+            return new OverflowAwareMutableLong(0L);
+        }
+
+        @Override
+        public void loadDocValues(LeafReader reader) throws IOException {
+            values = DocValues.getSortedNumeric(reader, columnName);
+        }
+
+        @Override
+        public void apply(RamAccounting ramAccounting,
+                          int doc,
+                          OverflowAwareMutableLong state) throws IOException {
+            if (values.advanceExact(doc) && values.docValueCount() == 1) {
+                state.add(values.nextValue());
+            }
+        }
+
+        @Override
+        public BigDecimal partialResult(RamAccounting ramAccounting,
+                                        OverflowAwareMutableLong state) {
+            if (state.hasValue()) {
+                var partialResult = state.value();
+                ramAccounting.addBytes(partialResult.unscaledValue().toByteArray().length);
```

Comment:
Same here.

I'm also not sure if `partialResult` is the right place for the accounting. `partialResult` is called at the end of the aggregation, long after the allocations and resizes already happened.

I think doing this in `initialState` or as needed would be safer

Reply:
I put into `partialResult` because we do not know how big the big decimal can be at the end. Would you suggest just to put some apprx. into `initialState` like for example 4 bytes?

Reply:
Yes I think its important to account that at earliest as possible, meaning accounting for the min size at `initialState` and adjusting it on `iterate`.